pkg/registry/apis/secret/secretkeeper/secretkeeper.go

```diff
@@ -3,43 +3,37 @@ package secretkeeper
 import (
 	"fmt"
 
+	secretv0alpha1 "github.com/grafana/grafana/pkg/apis/secret/v0alpha1"
 	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/grafana/grafana/pkg/registry/apis/secret/contracts"
 	"github.com/grafana/grafana/pkg/registry/apis/secret/secretkeeper/sqlkeeper"
 )
 
-// Service is the interface for secret keeper services.
-// This exists because OSS and Enterprise have different amounts of keepers available.
-type Service interface {
-	GetKeepers() (map[contracts.KeeperType]contracts.Keeper, error)
-}
-
 // OSSKeeperService is the OSS implementation of the Service interface.
 type OSSKeeperService struct {
-	tracer            tracing.Tracer
-	encryptionManager contracts.EncryptionManager
-	store             contracts.EncryptedValueStorage
+	systemKeeper *sqlkeeper.SQLKeeper
 }
 
+var _ contracts.KeeperService = (*OSSKeeperService)(nil)
+
 func ProvideService(
 	tracer tracing.Tracer,
 	store contracts.EncryptedValueStorage,
 	encryptionManager contracts.EncryptionManager,
-) (OSSKeeperService, error) {
-	return OSSKeeperService{
-		tracer:            tracer,
-		encryptionManager: encryptionManager,
-		store:             store,
-	}, nil
-}
-
-func (ks OSSKeeperService) GetKeepers() (map[contracts.KeeperType]contracts.Keeper, error) {
-	sqlKeeper, err := sqlkeeper.NewSQLKeeper(ks.tracer, ks.encryptionManager, ks.store)
+) (*OSSKeeperService, error) {
+	// TODO: rename to system keeper or something like that
+	systemKeeper, err := sqlkeeper.NewSQLKeeper(tracer, encryptionManager, store)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create sql keeper: %w", err)
 	}
 
-	return map[contracts.KeeperType]contracts.Keeper{
-		contracts.SQLKeeperType: sqlKeeper,
+	return &OSSKeeperService{
+		systemKeeper: systemKeeper,
 	}, nil
 }
+
+// Ignore the config, but we could use it to get the keeper type and then return the correct keeper.
+// Instantiation only happens on ProvideService ONCE.
+func (k *OSSKeeperService) KeeperForConfig(secretv0alpha1.KeeperConfig) (contracts.Keeper, error) {
```

Comment:
Also `GetKeepers()` is gone. Why? We always try to fetch a keeper per time, so we can encapsulate the available keeper list behind this service, without having to pass around the `map[KeeperType]Keeper`.

Especifically for OSS, this makes it nice because the `secretv0alpha1.KeeperConfig` will be `nil`, but we don't care because it is not used.